tests/integration/frameworks/test_frameworks.py

```diff
@@ -54,9 +54,13 @@ def test_generic_arguments(framework: types.ModuleType, test_model: FrameworkTes
     assert scaler.var_[0] == 4.75  # type: ignore (bad sklearn types)
 
     kwargs = test_model.save_kwargs.copy()
-    kwargs["signatures"] = {
-        "pytest-signature-rjM5": {"batchable": True, "batch_dim": (1, 0)}
-    }
+    default_meth = "pytest-signature-rjM5"
+    if test_model.model_signatures:
+        kwargs["signatures"] = test_model.model_signatures
+        meths = list(test_model.model_signatures.keys())
+    else:
+        kwargs["signatures"] = {default_meth: {"batchable": True, "batch_dim": (1, 0)}}
+        meths = [default_meth]
```

Comment:
cc @sauyon some changes for `FrameworkTest` for onnx or other more flexible frameworks.